### PR TITLE
Add Cl regime

### PR DESCRIPTION
**Pull request by pacots**
## Summary

Adds the initial Chile (`CL`) tax regime.

This PR focuses on core GOBL regime support for Chile: country metadata, IVA/VAT, RUT tax identity normalization and validation, invoice correction types, generated data, and a minimal example invoice.

Chile-specific SII/DTE behavior is deliberately left out of scope and would be better handled in a future addon.

## What's included

- `regimes/cl`: initial Chile regime package.
- Chile regime registration via `regimes/regimes.go`.
- Chile regime metadata:
  - `CL` country code
  - `CLP` currency
  - `America/Santiago` timezone
  - VAT as the tax scheme
- IVA (`VAT`) tax category:
  - 19% general rate
  - effective from 2003-10-01
  - sourced from Decreto Ley 825, Article 14, including the Ley 19.888 effective-date note
- Chilean RUT tax identity support:
  - normalization removes separators and optional `CL` prefix
  - lowercase `k` is normalized to `K`
  - validation checks length, numeric body, valid check digit character, and Chilean modulo-11 checksum
- Invoice correction types:
  - credit notes
  - debit notes
- Minimal Chile invoice example under `examples/cl/`.
- Generated regime, rules, schema, and example output data.
- CHANGELOG entry.

## Tests

Adds tests for:

- Chile regime metadata.
- RUT normalization.
- Valid RUTs with numeric, zero, and `K` check digits.
- Invalid RUT checksum.
- Invalid RUT characters.
- Invalid RUT length.
- Empty optional tax identity code behavior.
- Modulo-11 edge cases through public validation behavior.

Focused checks run locally:

```bash
go test ./regimes/cl/... -coverprofile=coverage
go tool cover -func=coverage
golangci-lint run ./regimes/cl/...
```

`go test ./regimes/cl/...` reports 94.7% coverage.

Generated data and example outputs were updated with:

```bash
go generate .
go test --update ./...
```

## Deliberately deferred to a future SII/DTE addon

These are real Chilean e-invoicing concerns, but they are format/platform-specific and do not belong in this initial regime layer:

- DTE XML generation/parsing
- SII transmission, certification, status checks, or API integration
- CAF, folios, TED, electronic stamps, or PDF417 behavior
- Chile-specific DTE document type mapping
- Additional/specific taxes, retentions, and sector-specific taxes
- Chile-specific invoice-level DTE validation rules
- Chile-specific rounding rules beyond existing GOBL/currency behavior

No Chile-specific scenario mappings are added in this initial PR.

## Sources

Official Chilean sources:

- BCN - Decreto Ley 825, Ley sobre Impuesto a las Ventas y Servicios  
  https://www.bcn.cl/leychile/navegar?idNorma=6369

- SII - Formato Documentos Tributarios Electronicos  
  https://www.sii.cl/factura_electronica/formato_dte.pdf

## Pre-Review Checklist

- [x] Opened this PR as a draft
- [x] Read the CONTRIBUTING.md guide.
- [x] Performed a self-review of my code.
- [x] Added thorough tests with at **least** 90% code coverage.
- [x] Modified or created example GOBL documents to show my changes in use, if appropriate.
- [x] Added links to the source of the changes in tax regimes or addons, either structured or in the comments.
- [x] Run `go generate .` to ensure that the Schemas and Regime data are up to date.
- [x] Reviewed and fixed all linter warnings.
- [x] Been obsessive with pointer nil checks to avoid panics.
- [x] Updated the CHANGELOG.md with an overview of my changes.
- [x] Marked this PR as ready for review.

And if you are part of the org:
- [ ] Requested a review from Copilot and fixed or dismissed (with a reason) all the feedback raised.
- [ ] Requested a review from @samlown.

### CODE REVIEW

File: regimes/cl/tax_identity.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/invopop/gobl/rules/is"
+	"github.com/invopop/gobl/tax"
+)
+
+func taxIdentityRules() *rules.Set {
+	return rules.For(new(tax.Identity),
+		rules.When(tax.IdentityIn(CountryCode),
+			rules.Field("code",
+				rules.AssertIfPresent("01", "invalid Chilean RUT tax identity code",
+					is.Func("valid", isValidTaxIdentityCode),
```
Comment (samlown) [inline regimes/cl/tax_identity.go]: Rather than a single identity check, it'd probably be useful to have specific assertions for each of the checks, so that its possible to get specific errors for invalid length, bad characters, invalid check digits, and the checksum. That could make them much easier to debug.

File: regimes/cl/tax_identity.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+func isValidTaxIdentityCode(value any) bool {
+	code, ok := value.(cbc.Code)
+	if !ok || code == "" {
+		return false
+	}
+	return validateTaxCode(code) == nil
+}
+
+func validateTaxCode(code cbc.Code) error {
```
Comment (samlown) [inline regimes/cl/tax_identity.go]: We tend to try to avoid embedding error message responses during validation as they are not accessible anywhere. (See previous comment).

**Review by samlown — CHANGES_REQUESTED** [review]
Thanks for this! Looks good. I'd suggest refactoring the assertions to provide more detailed fault messages.

File: regimes/cl/tax_identity.go (hunk 1 above)
Comment (pacots) [reply regimes/cl/tax_identity.go]: Thanks, that makes sense. I’ve split the RUT validation into separate assertions for length, numeric body, check digit, and checksum, and updated the tests to cover each case.

File: regimes/cl/tax_identity.go (hunk 2 above)
Comment (pacots) [reply regimes/cl/tax_identity.go]: Removed the internal error messages and replaced them with rule assertions.
